### PR TITLE
LPS-145113 - Add a site configuration for selecting specific icon packs

### PR DESCRIPTION
**Pull request by bryceosterhaus**
https://issues.liferay.com/browse/LPS-145113

This gives site's the ability to select their own icon packs to be available for their site specific spritemap

### CODE REVIEW

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:test:relevant

File: modules/apps/frontend-icons/frontend-icons-web/src/main/resources/META-INF/resources/js/site_settings/index.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+				...acc,
+				[packName]: {
+					...icons[packName],
+					icons: icons[packName].icons
+						.filter((icon) => Object.keys(icon).length)
+						.filter((icon) =>
+							icon.name
+								.toLowerCase()
+								.includes(searchQuery.toLocaleLowerCase())
+						),
```
Comment (diegonvs) [inline modules/apps/frontend-icons/frontend-icons-web/src/main/resources/META-INF/resources/js/site_settings/index.js:48]: Hey @bryceosterhaus, why not:

```suggestion
						
						.filter((icon) =>
						        Object.keys(icon).length &&
							icon.name
								.toLowerCase()
								.includes(searchQuery.toLocaleLowerCase())
						),
```

File: modules/apps/frontend-icons/frontend-icons-web/src/main/resources/META-INF/resources/js/site_settings/index.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+				/>
+			</label>
+
+			<ClayPanel.Group className="mt-4">
+				<span className="form-control-label-text">
+					{Liferay.Language.get('available-icon-packs')}
+				</span>
+
+				{iconPackNames
+					.filter((name) => selectedPacks.includes(name))
```
Comment (diegonvs) [inline modules/apps/frontend-icons/frontend-icons-web/src/main/resources/META-INF/resources/js/site_settings/index.js:91]: Hey @bryceosterhaus, how many items are expected to be at selectedPacks and inside iconPackNames? filter -> includes is quadratic. Maybe we can use a dict/Map on `initialSelectedPacks`

Comment (bryceosterhaus) [reply modules/apps/frontend-icons/frontend-icons-web/src/main/resources/META-INF/resources/js/site_settings/index.js:91]: @diegonvs I wouldn't really expect more than 10, even that would be large though, so I think this should be fine.

File: modules/apps/frontend-icons/frontend-icons-web/src/main/resources/META-INF/resources/js/site_settings/index.js (hunk 1 above)
Comment (bryceosterhaus) [reply modules/apps/frontend-icons/frontend-icons-web/src/main/resources/META-INF/resources/js/site_settings/index.js:48]: hmm, I actually don't really remember why I added it this way. I will take another look

**Comment by bryceosterhaus** [comment]
@diegonvs I am gonna forward and address your issues in a follow up PR when we clean up the UI part of this epic.

**Comment by bryceosterhaus** [comment]
ci:forward

**Pull request closed without merging**